### DISCUSSION TITLE
Can someone please explain the names in Tasmota - I am confused

### DISCUSSION THREAD

**Discussion by wayner9** [arendst/Tasmota#20323] [Q&A]
In Tasmota there are a bunch of places to enter names.  When you go to the Config- Other page there is a name that is in the template, Device Name and Friendly Name 1.
Then in the MQTT config there is the Client, Topic and Full Topic

How should I be using these.  Should some of these retain the original device name, like "Sonoff S31"?  Where does it makes sense to use other things for the name, like "Kitchen East Wall" or "unRAID server".

I know that it may not really matter, but I am just trying to understand if there is a naming hierarchy, and where these names show up.

**Comment by sfromis** [comment]
The names are there to provide you with option for how you want it set it up. There is no "should" about any of the default names, in principle I'd even suggest you to make your own naming schemes.

What's important is where the names are used, and this is reflected in what they are called, like when:

topic refers to the topic in MQTT, and is not mean as a text name to talk about.
full topic is a mask for what becomes the topic string in MQTT, by default
`%prefix%/%topic%/` resulting in MQTT topics like `stat/yourdevice/POWER`
MQTT client name is less important, but should always be unique to the broker, not used more than there. I'd suggest leaving it as default, unless you like browsing broker logs.

device name is OTOH thought as a "more readable" name for what the Tasmota device is.
friendly name is similar, but there can be multiple (like multiple relays), and is used with voice assistants and HA.
The template name is intended to match the hardware, but you are free to do whatever.
